lib/utility/src/WatchdogWrapper.cpp

```diff
@@ -0,0 +1,35 @@
+
+#include "WatchdogWrapper.h"
+
+#include <string>
+
+#include "Logger.h"
+#include "ResetReason.h"
+#include "mbed.h"
+
+namespace Utility {
+
+void WatchdogWrapper::startWatchdog(std::chrono::milliseconds countdown_ms /*= 5000ms*/,
+                                    std::chrono::milliseconds pet_ms /*= 1000ms*/) {
+  uint32_t countdown_uint32 = countdown_ms.count();
+  Watchdog &watchdog        = Watchdog::get_instance();
+  watchdog.start(countdown_uint32);
+  Thread pet_thread;
```

Comment:
Since `pet_thread` is created within the `startWatchdog()` function, its scope is limited to this function and will be destroyed when the function exits, but you want it to persist since this thread runs continuously. You should instead make `pet_thread` a private static variable of the WatchdogWrapper class